### DISCUSSION TITLE
Schema model for invoice structure help

### DISCUSSION THREAD

**Discussion by gmcamposano** [prisma/prisma#16151] [Q&A] [answered]
I want to be able to have multipe invoices that have multiple products. How can i achieve this?
invoice 1: product1, product2
invoice 2: product1, product3, product5
..
```
model Product {
  id        Int      @id @default(autoincrement())
  name      String
  price     Int
  invoice   Invoice[]
}

model Invoice {
  id        Int      @id @default(autoincrement())
  product_id Int
  product   Product  @relation(fields: [product_id], references: [id])
  total     Int
}
```

**Comment by nurul3101** [comment, marked as answer]
Hello @gmcamposano 👋 
It seems you are looking for many to many relations.
Could you try this schema?
```prisma
model Product {
  id      Int       @id @default(autoincrement())
  name    String
  price   Int
  invoice Invoice[]
}

model Invoice {
  id      Int       @id @default(autoincrement())
  product Product[]
  total   Int
}
```
There is an [implicit many-to-many relation](https://www.prisma.io/docs/concepts/components/prisma-schema/relations/many-to-many-relations#implicit-many-to-many-relations) between Product and Invoice. Another alternative could be to create an [explicit many-to-many relations](https://www.prisma.io/docs/concepts/components/prisma-schema/relations/many-to-many-relations#explicit-many-to-many-relations).
Please let me know in case you run into any issues in trying this approach.

**Reply by gmcamposano** [reply]
I later on discovered in the docs that prisma gives us the ability to do nested inserts. Sorry about that.

**Reply by nurul3101** [reply]
No worries, I am glad to hear that you were able to find the solution.

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that have already been marked as answered but remain open.

If this discussion still requires further input or clarification, feel free to reopen it or start a new one with updated details. Your contributions are invaluable to the community, and we’re here to help!

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and ongoing support of the Prisma community!
